src/hotspot/share/oops/fieldStreams.hpp

```diff
@@ -131,7 +131,7 @@ class FieldStreamBase : public StackObj {
   // bridge to a heavier API:
   fieldDescriptor& field_descriptor() const {
     fieldDescriptor& field = const_cast<fieldDescriptor&>(_fd_buf);
-    field.reinitialize(field_holder(), _index);
+    field.reinitialize(field_holder(), _fi_buf);
```

Comment:
`_fi_buf` - > `to_FieldInfo()`? Reads better, and I expect it to fully inline without any performance loss.